src/main/java/seedu/address/logic/commands/project/EditTaskCommand.java

```diff
@@ -0,0 +1,253 @@
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Project project = model.getFilteredProjectList().get(0);
+        List<Task> lastShownList = project.getFilteredTaskList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        Task taskToEdit = lastShownList.get(index.getZeroBased());
+        Task editedTask = createEditedTask(taskToEdit, editTaskDescriptor);
+
+        lastShownList.remove(taskToEdit);
+        lastShownList.add(editedTask);
+
+        return new CommandResult(String.format(MESSAGE_EDIT_TASK_SUCCESS, editedTask));
+    }
+
+    /**
+     * Creates and returns a {@code Task} with the details of {@code taskToEdit}
+     * edited with {@code editTaskDescriptor}.
+     */
+    private static Task createEditedTask(Task taskToEdit, EditTaskDescriptor editTaskDescriptor) {
+        assert taskToEdit != null;
+
+        String updatedTaskName = editTaskDescriptor.getTaskName().orElse(taskToEdit.getTaskName());
+        Deadline updatedDeadline = editTaskDescriptor.getDeadline().orElse(taskToEdit.getDeadline());
+        Double updatedProgress = editTaskDescriptor.getProgress().orElse(taskToEdit.getProgress());
+        Boolean updatedIsDone = editTaskDescriptor.getIsDone().orElse(taskToEdit.isDone());
+        String updatedTaskDescription = editTaskDescriptor.getTaskDescription()
+                .orElse(taskToEdit.getDescription());
+        Set<Participation> updatedAssignees = editTaskDescriptor.getAssignees().orElse(
+                taskToEdit.getAssignees());
+
+        Task updatedTask = new Task(updatedTaskName, updatedTaskDescription, updatedDeadline,
+                updatedProgress, updatedIsDone);
+        updatedTask.getAssignees().addAll(updatedAssignees);
+
+        return updatedTask;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditTaskCommand)) {
+            return false;
+        }
+
+        // state check
+        EditTaskCommand e = (EditTaskCommand) other;
+        return index.equals(e.index)
+                && editTaskDescriptor.equals(e.editTaskDescriptor);
+    }
+
+    /**
+     * Stores the details to edit the task with. Each non-empty field value will replace the
+     * corresponding field value of the task.
+     */
+    public static class EditTaskDescriptor {
+        String taskName;
+        String description;
+        LocalDate publishDate;
+        Deadline deadline;
+        Double progress;
+        Boolean isDone;
+        Set<Participation> assignees;
+
+        public EditTaskDescriptor() {}
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code taskTags} is used internally.
```

Comment:
This lines does not match the function. Especially the @code taskTags. Is there any missing refactor for this comment?